File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    start = 0
    for i in range(0, (len(t) - len(p)) + 1):
        if int(p) >= int(t[start: start + len(p)]):
            answer +=1
        start += 1
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int len = p.length();
        long pValue = Long.parseLong(p); 
        for (int i = 0; i <= t.length() - len; i++) {
            String chunk = t.substring(i, i + len);

            if (Long.parseLong(chunk) <= pValue) {
                answer++;
            }
        }
        return answer;
    }
}